libraries/standard/http/utest/http_utest.c

```diff
@@ -248,4 +344,297 @@ void test_Http_InitializeRequestHeaders_insufficient_memory()
+                                         10 /* rangeStart */,
+                                         5 /* rageEnd */ );
+    TEST_ASSERT_EQUAL( HTTP_INVALID_PARAMETER, retCode );
+
+    /* rangeStart is negative but rangeStart is non-End of File. */
+    tearDown();
+    testHeaders.pBuffer = &testBuffer[ 0 ];
+    retCode = HTTPClient_AddRangeHeader( &testHeaders,
+                                         -10 /* rangeStart */,
+                                         HTTP_RANGE_REQUEST_END_OF_FILE + 1 /* rageEnd */ );
+    TEST_ASSERT_EQUAL( HTTP_INVALID_PARAMETER, retCode );
+    tearDown();
+    testHeaders.pBuffer = &testBuffer[ 0 ];
+    retCode = HTTPClient_AddRangeHeader( &testHeaders,
+                                         -50 /* rangeStart */,
+                                         -10 /* rageEnd */ );
+    TEST_ASSERT_EQUAL( HTTP_INVALID_PARAMETER, retCode );
+}
+
+/**
+ * @brief Test Insufficient memory failure when the buffer has one less byte than required.
+ */
+void test_Http_AddRangeHeader_Insufficient_Memory( void )
+{
+    setupBuffersWithPreexistingHeader( &testHeaders,
+                                       testBuffer,
+                                       sizeof( testBuffer ),
+                                       &expectedHeaders,
+                                       PREEXISTING_HEADER_DATA );
+    size_t preHeadersLen = testHeaders.headersLen;
+    testRangeStart = 5;
+    testRangeEnd = 10;
+
+    /* Update the expected header with the complete the range request header
+     * to determine the total required size of the buffer. */
+    addRangeToExpectedHeaders( &expectedHeaders,
+                               "5-10" /*expected range*/,
+                               true );
+
+    /* Change the input headers buffer size to be one byte short of the required
+     * size to add Range Request header. */
+    testHeaders.bufferLen = expectedHeaders.dataLen - 1;
+
+    /* As the call to the API function is expected to fail, we need to store a
+     * local copy of the input headers buffer to verify that the data has not changed
+     * after the API call returns. Thus, overwrite the expected headers buffer with the
+     * copy of the complete input headers buffer to use for verification later. */
+    TEST_ASSERT_GREATER_OR_EQUAL( testHeaders.bufferLen, sizeof( expectedHeaders.buffer ) );
+    memcpy( expectedHeaders.buffer, testHeaders.pBuffer, testHeaders.bufferLen );
+
+    retCode = HTTPClient_AddRangeHeader( &testHeaders,
+                                         testRangeStart,
+                                         testRangeEnd );
+    TEST_ASSERT_EQUAL( HTTP_INSUFFICIENT_MEMORY, retCode );
+    /* Verify the headers input parameter is unaltered. */
+    TEST_ASSERT_EQUAL( preHeadersLen, testHeaders.headersLen );
+    TEST_ASSERT_EQUAL( expectedHeaders.dataLen - 1, testHeaders.bufferLen );
+    TEST_ASSERT_EQUAL_MEMORY( expectedHeaders.buffer,
```

Comment:
This is nice, I straight up wrote a for-loop in previous tests to check this haha.